### PR TITLE
Deprecate schedule prediction file config keyword

### LINKED ISSUE #4288: Check if SCHEDULE_PREDICTION_FILE can/should have additional arguments in the config file
Currently, the key SCHEDULE_PREDICTION_FILE allows the additional arguments 
 * PARAMETER
 * INIT_FILES

It is not clear if they should be allowed or not. If not please remove the code that handles them.

### PR DESCRIPTION
**Pull request by valentin-krasontovitsch**
**Issue**
Resolves #4288

**Approach**
We deprecate the keyword.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by valentin-krasontovitsch** [comment]
jenkins, retest this please

**Comment by andreas-el** [comment]
We are using the `SCHEDULE_PREDICTION_FILE` parameter in ert files found in the tests.
Should we update those since we deprecate the keyword? Or is that left to when we actually remove the support for the keyword?

```
./test-data/configuration_tests/ensemble_config.ert:18:SCHEDULE_PREDICTION_FILE input/schedule.sch PARAMETER:MULTFLT.TXT INIT_FILES:fields/permx%d.grdecl
./test-data/configuration_tests/ecl_config.ert:5:SCHEDULE_PREDICTION_FILE        input/schedule.sch
```

**Comment by oysteoh** [comment]
> We are using the `SCHEDULE_PREDICTION_FILE` parameter in ert files found in the tests. Should we update those since we deprecate the keyword? Or is that left to when we actually remove the support for the keyword?
> 
> ```
> ./test-data/configuration_tests/ensemble_config.ert:18:SCHEDULE_PREDICTION_FILE input/schedule.sch PARAMETER:MULTFLT.TXT INIT_FILES:fields/permx%d.grdecl
> ./test-data/configuration_tests/ecl_config.ert:5:SCHEDULE_PREDICTION_FILE        input/schedule.sch
> ```

I guess leaving them until actual removal is the best approach to avoid breaking something which might be in use ( even though deprecated ) ?

**Review by andreas-el — CHANGES_REQUESTED** [review]

File: src/clib/lib/config/config_keywords.cpp [hunk 1]
```diff
@@ -73,7 +73,6 @@ static void add_history_source_keyword(config_parser_type *config_parser) {
     auto item = add_single_arg_keyword(config_parser, HISTORY_SOURCE_KEY);
 
     stringlist_type *argv = stringlist_alloc_new();
-    stringlist_append_copy(argv, "SCHEDULE");
```
[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply src/clib/lib/config/config_keywords.cpp:76]: yes it is indeed!

File: src/clib/lib/config/config_keywords.cpp [hunk 2]
```diff
@@ -206,6 +206,9 @@ add_schedule_prediction_file_keyword(config_parser_type *config_parser) {
     /* scedhule_prediction_file   filename  <parameters:> <init_files:> */
     config_schema_item_set_argc_minmax(item, 1, 3);
     config_schema_item_iset_type(item, 0, CONFIG_EXISTING_PATH);
+    config_parser_deprecate(
+        config_parser, SCHEDULE_PREDICTION_FILE_KEY,
+        "The SCHEDULE_PREDICTION_FILE config key is deprecated.");
```
[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply src/clib/lib/config/config_keywords.cpp:211]: yes, that sounds reasonable. assuming we always use the deprecation in the same context as in this example (the item was just added to the parser, hence we know it exists), the code of `config_parser_deprecate` (which checks if the item exists, and util aborts if not) is not really necessary at all. good catch, thank you!

File: docs/reference/configuration/keywords.rst [hunk 3]
```diff
@@ -1503,9 +1503,7 @@ but required when installing ERT at a new site.
 .. _schedule_prediction_file:
 .. topic:: SCHEDULE_PREDICTION_FILE
 
-        This is the name of a schedule prediction file. It can contain %d to get
-        different files for different members. Observe that the ECLIPSE datafile
-        should include only one schedule file, even if you are doing predictions.
```
Comment (sondreso) [inline docs/reference/configuration/keywords.rst:1508]: I would keep the documentation until we actually remove the keyword, but say "This keyword is deprecated and will be removed." at the top.

File: src/clib/lib/config/config_keywords.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply src/clib/lib/config/config_keywords.cpp]: it is done.

File: docs/reference/configuration/keywords.rst (hunk 3 above)
Comment (valentin-krasontovitsch) [reply docs/reference/configuration/keywords.rst]: very well! it is done.

File: src/clib/lib/config/config_parser.cpp [hunk 4]
```diff
@@ -663,12 +663,3 @@ void config_install_message(config_parser_type *config, const char *kw,
     hash_insert_hash_owned_ref(config->messages, kw,
                                util_alloc_string_copy(message), free);
 }
-
-void config_parser_deprecate(config_parser_type *config, const char *kw,
```
Comment (andreas-el) [inline src/clib/lib/config/config_parser.cpp:667]: There is still code related to this present;
config_parser.hpp
```
void config_parser_deprecate(config_parser_type *config, const char *kw,
                             const char *msg);
```
Should be removed also.

**Review by andreas-el — CHANGES_REQUESTED** [review]

File: src/clib/lib/config/config_parser.cpp (hunk 4 above)
Comment (valentin-krasontovitsch) [reply src/clib/lib/config/config_parser.cpp:667]: ahh very good, of course, the header file. i didn't even grep for the function... thanks again ^^

Comment (valentin-krasontovitsch) [reply src/clib/lib/config/config_parser.cpp:667]: it is done.

**Review by andreas-el — APPROVED** [review]
:suspect: approved!

**Pull request merged**